cpp/log-levels/log_levels.cpp

```
@@ -0,0 +1,27 @@
+}
+
+std::string log_level(std::string line) {
+    // return the log level
+	const auto pos_end = line.find("]");
+	auto log_lvl = line.substr(1, pos_end - 1);
+	std::transform(begin(log_lvl), end(log_lvl), begin(log_lvl),
+				   [](auto c) { return std::toupper(c); });
+	return log_lvl;
+}
+
+std::string reformat(std::string line) {
```

Cppcheck (2.10-2)

[performance] Function parameter 'line' should be passed by const reference.

(passedByValue)